analytics/src/test/java/com/segment/analytics/QueueFileTest.kt

```diff
@@ -0,0 +1,953 @@
+
+            for (i in value.indices) {
+                Assertions.assertThat(value[i])
+                    .isEqualTo((blockNum + 1).toByte())
+                    .`as`("Block " + (blockNum + 1) + " corrupted at byte index " + i)
+            }
+        }
+
+        queue.close()
+    }
+
+    /**
+     * Exercise a bug where wrapped elements were getting corrupted when the QueueFile was forced to
+     * expand in size and a portion of the final Element had been wrapped into space at the beginning
+     * of the file - if multiple Elements have been written to empty buffer space at the start does
+     * the expansion correctly update all their positions?
+     */
+    @Test
+    @Throws(IOException::class)
+    fun testFileExpansionCorrectlyMovesElements() {
+        val queue = QueueFile(file)
+
+        // Create test data - 1k blocks marked consecutively 1, 2, 3, 4 and 5.
+        val values = arrayOfNulls<ByteArray>(5)
+        for (blockNum in values.indices) {
+            values[blockNum] = ByteArray(1024)
+            for (i in 0 until (values[blockNum]!!.size)) {
+                values[blockNum]!![i] = (blockNum + 1).toByte()
+            }
+        }
+
+        // smaller data elements
+        val smaller = arrayOfNulls<ByteArray>(3)
+        for (blockNum in smaller.indices) {
+            smaller[blockNum] = ByteArray(256)
+            for (i in smaller[blockNum]!!.indices) {
+                smaller[blockNum]!![i] = (blockNum + 6).toByte()
+            }
+        }
+
+        // First, add the first two blocks to the queue, remove one leaving a
+        // 1K space at the beginning of the buffer.
+        queue.add(values[0])
+        queue.add(values[1])
+        queue.remove()
+        // The trailing end of block "4" will be wrapped to the start of the buffer.
+        queue.add(values[2])
+        queue.add(values[3])
+
+        // Now fill in some space with smaller blocks, none of which will cause
+        // an expansion.
+        queue.add(smaller[0])
+        queue.add(smaller[1])
+        queue.add(smaller[2])
+
+        // Cause buffer to expand as there isn't space between the end of the
+        // smaller block "8" and the start of block "2".  Internally the queue
+        // should cause all of tbe smaller blocks, and the trailing end of
+        // block "5" to be moved to the end of the file.
+        queue.add(values[4])
+
+        val expectedBlockNumbers = byteArrayOf(2, 3, 4, 6, 7, 8, 5)
+
+        // Make sure values are not corrupted, specifically block "4" that wasn't
+        // being made contiguous in the version with the bug.
+        for (expectedBlockNumber in expectedBlockNumbers) {
+            val value = queue.peek()
+            queue.remove()
+
+            for (i in value.indices) {
+                Assertions.assertThat(value[i])
+                    .isEqualTo(expectedBlockNumber)
+                    .`as`("Block $expectedBlockNumber corrupted at byte index $i")
```

Comment:
is there a different method than `as` that we can use here?

Reply:
Yup! You can use "to" instead since "as" in Kotlin is used for type conversions